bin/agent-data-plane/src/main.rs

```diff
@@ -93,9 +96,11 @@ async fn run(started: Instant) -> Result<(), GenericError> {
     let env_provider =
         ADPEnvironmentProvider::from_configuration(&configuration, &component_registry, &health_registry).await?;
 
+    let status_configuration = DatadogStatusFlareConfiguration::from_configuration(&configuration).await?;
```

Comment:
I have it in `run` because `create_topology` isn't async. And the reason `DatadogStatusFlareConfiguration::from_configuration` is async is because of `RemoteAgentClient::from_configuration`